sky/optimizer.py

```diff
@@ -390,16 +390,27 @@ def get_reservations_available_resources(
                 node_resources_reprs = ', '.join(f'{node.num_nodes}x ' +
                                                  r.repr_with_region_zone
                                                  for r in node.resources)
+                hints_concat = '\n'.join([
+                    f'{bold}Resource: {repr(resource)}{reset}\n' +
+                    '\n'.join(hint_list)
+                    for resource, hint_list in resource_hints.items()
+                    if hint_list
+                ])
+                hints_formatted = '\n'.join(
+                    map(lambda r: f'        {r}', hints_concat.split('\n')))
+                resource_hints_string = (
+                    f'Hint 2: Check Per Resource Hint\n{hints_formatted}'
+                    if hints_formatted else '')
                 error_msg = (
                     f'{source_hint.capitalize()} does not contain any '
                     f'instances satisfying the request: '
                     f'{node_resources_reprs}.'
                     f'\nTo fix: relax or change the '
                     f'resource requirements.{fuzzy_candidates_str}\n\n'
-                    f'Hint: {bold}sky show-gpus{reset} '
+                    f'Hint 1: {bold}sky show-gpus{reset} '
                     'to list available accelerators.\n'
-                    f'      {bold}sky check{reset} to check the enabled '
-                    'clouds.')
+                    f'        {bold}sky check{reset} to check the enabled '
+                    f'clouds.\n{resource_hints_string}')
```

Comment:
```suggestion
                    'clouds.\n'
                    '{resource_hints_string}')
```
`clouds.\n` does becomes a one word line, but I do generally prefer the string split to align with newlines.